olm-catalog/serverless-operator/operator.environment.yaml

```diff
@@ -0,0 +1,24 @@
+---
+openshift:
```

Comment:
I'm trying to find any system in how these variables are named. It looks a bit random.
Perhaps we could follow how the repositories are really named but then I'm wondering how to add any random information.
I'd say this file needs some well-defined structure or needs to be more flat. But I don't know what structure to suggest right now.

Reply:
I basically rewritten the variables that we were using to be nested.

```java
static final String OPENSHIFT_KNATIVE_SERVING_BRANCH = "release-v0.13.3"
static final String OPENSHIFT_KNATIVE_SERVING_CI_PROMOTION_NAME = "knative-v0.13.3"
```
bacame:

```yaml
openshift:
  knative:
    serving:
      branch: release-v0.13.3
      ci-promotion-name: knative-v0.13.3
```

with the idea that you'll fetch them with something like:

```bash
$ opskope registry.example.org/openshift/serverless:v1.8.0 
    metadata-get openshift.knative.serving.branch
```

Or maybe itarete on some keys, like:

```bash
$ opskope registry.example.org/openshift/serverless:v1.8.0 
    metadata-foreach --output json openshift.serverless.upgradepaths | ...
```

Reply:
I basically rewritten the variables that we were using to be nested.

```java
static final String OPENSHIFT_KNATIVE_SERVING_BRANCH = "release-v0.13.3"
static final String OPENSHIFT_KNATIVE_SERVING_CI_PROMOTION_NAME = "knative-v0.13.3"
```
bacame:

```yaml
openshift:
  knative:
    serving:
      branch: release-v0.13.3
      ci-promotion-name: knative-v0.13.3
```

with the idea that you'll fetch them with something like:

```bash
$ opskope registry.example.org/openshift/serverless:v1.8.0 
    metadata-get openshift.knative.serving.branch
```

Or maybe itarete on some keys, like:

```bash
$ opskope registry.example.org/openshift/serverless:v1.8.0 
    metadata-foreach --output json openshift.serverless.upgradepaths | ...
```

Reply:
ok. Thanks. I was thinking about two ways to give it more well-defined structure:
1) Follow the repository hierarchy and names as we have them on github:
```
---
sources:
  openshift:
    knative-serving:
      branch: release-v0.13.3
      cirr-promotion-name: knative-v0.13.3
    knative-eventing:
      branch: release-v0.13.0
      ci-promotion-name: knative-v0.13.0
    knative-eventing-contrib:
      branch: release-v0.15.0
      ci-promotion-name: knative-v0.15.0
    knative-client:
      branch: openshift-v0.13.2
      url: https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/
    tektoncd-pipeline-operator:
      channel: preview
  openshift-knative:
    knative-kafka-operator:
      branch: v0.15.0
    knative-camel-operator:
      branch: master
    operator:
      branch: master
custom:
  ???

```

2) Or use a flat structure like this:
```
openshift.knative-serving.branch: release-v0.13.3
openshift.knative-serving.ci-promotion-name: knative-v0.13.3
...
```

It would be good to see any other items that we want to put in this file.

Reply:
I would prefer the first, nested, proposition as there are tools already to parse such structure. I'm thinking of [jsonpath](https://support.smartbear.com/alertsite/docs/monitors/api/endpoint/jsonpath.html).